machinery-helper/machinery_helper_test.go

```diff
@@ -0,0 +1,191 @@
+    t.Errorf("parseRpmLine('%s') file type = '%s', want '%s'", line, link_target, expected_link_target)
+  }
+}
+
+func TestParseRpmLineFile(t *testing.T) {
+  line := "-rw-r--r--    1 root    root                 18234080 Mar 31 11:40 /usr/lib64/libruby2.0-static.a"
+
+  expected_file_type := "-"
+  expected_file_name := "/usr/lib64/libruby2.0-static.a"
+  expected_link_target := ""
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestParseRpmLineDir(t *testing.T) {
+  line := "drwxr-xr-x    2 root    root                        0 Mar 31 11:45 /usr/include/ruby-2.0.0/x86_64-linux/ruby"
+
+  expected_file_type := "d"
+  expected_file_name := "/usr/include/ruby-2.0.0/x86_64-linux/ruby"
+  expected_link_target := ""
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestParseRpmLineLink(t *testing.T) {
+  line := "lrwxrwxrwx    1 root    root                       19 Mar 31 11:45 /usr/lib64/libruby2.0.so -> libruby2.0.so.2.0.0"
+
+  expected_file_type := "l"
+  expected_file_name := "/usr/lib64/libruby2.0.so"
+  expected_link_target := "libruby2.0.so.2.0.0"
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestParseRpmLineFileSpaces(t *testing.T) {
+  line := "-rw-r--r--    1 root    root                    61749 Jun 26 01:56 /usr/share/kde4/templates/kipiplugins_photolayoutseditor/data/templates/a4/h/Flipping Tux Black.ple"
+
+  expected_file_type := "-"
+  expected_file_name := "/usr/share/kde4/templates/kipiplugins_photolayoutseditor/data/templates/a4/h/Flipping Tux Black.ple"
+  expected_link_target := ""
+
+  doTestParseRpmLine(t, line, expected_file_type, expected_file_name, expected_link_target)
+}
+
+func TestAddImplicitlyManagedDirs(t *testing.T) {
+  files_original := map[string]string{
+    "/abc/def/ghf/somefile": "",
+    "/zzz": "/abc/def",
+  }
+  dirs_original := map[string]bool{
+    "/abc/def": true,
+  }
+  dirs_expected := map[string]bool{
+    "/abc": false,
+    "/abc/def": true,
+    "/abc/def/ghf": false,
+    "/zzz": false,
+  }
+
+  dirs := dirs_original
+
+  addImplicitlyManagedDirs(dirs, files_original)
+
+  if !reflect.DeepEqual(dirs, dirs_expected) {
+    t.Errorf("addImplicitlyManagedDirs('%v') = '%v', want '%v'", dirs_original, dirs, dirs_expected)
+  }
+}
+
+func TestAssembleJson(t *testing.T) {
+  unmanaged_files_map := map[string]string{
+    "name": "/usr/share/go_rulez", "type":"file",
+  }
+  want := `{
+   "extracted": false,
+   "files": {
+     "name": "/usr/share/go_rulez",
+     "type": "file"
+   }
+ }`
+
+  json := assembleJson(unmanaged_files_map)
+  if !reflect.DeepEqual(json, want) {
+    t.Errorf("assembleJson() = '%v', want '%v'", json, want)
+  }
+}
+
+func TestRespectManagedDirsInUnmanagedDirs(t *testing.T) {
+  unmanaged_files := make(map[string]string)
+  want_unmanaged_files := make(map[string]string)
```

Comment:
don't use underscores in Go names; var want_unmanaged_files should be wantUnmanagedFiles